glusterd2/volgen/store-utils.go

```diff
@@ -14,11 +15,27 @@ var (
 	volfilePrefix = "volfiles/"
 )
 
-func save(name string, content string) error {
+// SaveToStore saves the volfile content to etcd store
+func SaveToStore(name string, content string) error {
```

Comment:
I haven't went through the PR yet. Is it possible for volgen code to not save the volfile in store but to just return the content as a string ? Volfile generation is always synchronous i.e triggered by a user operation. Hence, storing can be done as single etcd txn that is run in a txn step. This single etcd txn will store volfile and volinfo as a atomic operation into etcd.

Reply:
As discussed earlier, only client volfiles are saved in store. So number of etcd transaction is limited to 1 per Volume Info changes. `SaveToStore` is wrapper around Volfile generation function, it can be easily changed to atomic operation as you suggested. I will look into this.